packages/session-replay-browser/src/events/events-idb-store.ts

```diff
@@ -33,8 +33,16 @@ export interface SessionReplayDB extends DBSchema {
 
 export const keyValDatabaseExists = function (): Promise<IDBDatabase | void> {
   const globalScope = getGlobalScope();
-  return new Promise((resolve) => {
-    if (globalScope) {
+  return new Promise((resolve, reject) => {
+    if (!globalScope) {
+      return reject(new Error('Global scope not found'));
+    }
+
+    if (!globalScope.indexedDB) {
+      return reject(new Error('IndexedDB is not available in the global scope'));
```

Comment:
I don't remember if this was discussed, but what were the reasons for this to not be available? is iframe the main one?

Reply:
There are a few security cases where indexeddb might be blocked from being used like CSP, iframe, browser extensions, etc. I think it's good to have the error handling here and provide some context in the docs on why it could be the case.

Reply:
SG. and just for context, when this fails, the SDK does not capture right (at least on this specific page)?

Reply:
In this context, this function is called to transition the old IDB store to the new store. I believe this is for the case where a user might open a new tab that has the newer version of the SDK and the stored events will transition over.